move/its/sources/service.move

```diff
@@ -368,24 +367,45 @@ module its::service {
     #[test]
     fun test_deploy_remote_interchain_token() {
         let mut its = its::its::new_for_testing();
+        let token_name = string::utf8(b"Name");
+        let token_symbol = ascii::string(b"Symbol");
+        let token_decimals = 10;
+        let remote_decimals = 12;
 
         let coin_info = its::coin_info::from_info<COIN>(
-            string::utf8(b"Name"),
-            ascii::string(b"Symbol"),
-            10,
-            12,
+            token_name,
+            token_symbol,
+            token_decimals,
+            remote_decimals,
         );
         let coin_management = its::coin_management::new_locked();
 
         let token_id = register_coin(&mut its, coin_info, coin_management);
         let destination_chain = ascii::string(b"Chain Name");
-        deploy_remote_interchain_token<COIN>(&mut its, token_id, destination_chain);
+        let message_ticket = deploy_remote_interchain_token<COIN>(&mut its, token_id, destination_chain);
+        
+        let mut writer = abi::new_writer(6);
+
+        writer
+            .write_u256(MESSAGE_TYPE_DEPLOY_INTERCHAIN_TOKEN)
+            .write_u256(token_id.to_u256())
+            .write_bytes(*token_name.as_bytes())
+            .write_bytes(*token_symbol.as_bytes())
+            .write_u256((token_decimals as u256))
+            .write_bytes(vector::empty());
+
+        assert!(message_ticket.source_id() == its.channel().to_address(), 0);
```

Comment:
expose `MessageTicket::new()` in tests so you can compare the two tickets directly maybe?

Reply:
@Foivos not addressed